cfme/tests/cli/test_appliance_cli.py

```diff
@@ -199,3 +202,35 @@ def is_failover_started(appliance):
     wait_for(is_failover_started, func_args=[apps[2]], timeout=450, handle_exception=True)
     apps[2].wait_for_evm_service()
     apps[2].wait_for_web_ui()
+
+
+def test_appliance_console_cli_configure_ipa(configured_appliance):
+    auth_provider_conf = auth_data.auth_providers.freeipa01
+    auth_provider_creds = credentials.get(auth_provider_conf.credentials)
+    result = configured_appliance.ssh_client.run_command(
+        (
+            "appliance_console_cli --ipaserver={ipaserver} --ipaprincipal={ipaprincipal} "
+            "--ipapassword={ipapassword} --ipadomain={ipadomain} --iparealm={iparealm}"
+        ).format(ipaserver=auth_provider_conf.host1,
+                 ipaprincipal=auth_provider_creds.ipaprincipal,
+                 ipapassword=auth_provider_creds.bind_password,
+                 ipadomain=auth_provider_conf.ipadomain,
+                 iparealm=auth_provider_conf.iparealm)
+    )
+    assert result.success
+    assert "Installation failed." not in result.output
+
+    view, _ = wait_for(lambda: navigate_to(configured_appliance.server, 'Authentication'),
+                       handle_exception=True, delay=10, timeout=120)
+    view.form.auth_mode.fill('External (httpd)')
+    view.form.httpd_role.fill("x")
+    view.save.click()
+
+    login_page = navigate_to(configured_appliance.server, 'LoginScreen')
+    ipa_users = [user for user in auth_data.test_data.test_users
+                 if 'freeipa01' in user.providers]
+    user = ipa_users[0]
```

Comment:
When testing login as different users, its best (required pretty much) to use the context manager.  Please see existing auth tests (cfme/tests/integration/test_cfme_auth) for use of this context manager with a user object. It sets the appliance user to the given user object, and any login/logout done within the context is done as the given user. At exit of the context, the admin user is restored as the default for the appliance.

Without the context controlling this, a failure within these statements would leave the appliance logged in as the non-admin user, impacting subsequent tests.